internal/tss/utils/transaction_service.go

```diff
@@ -0,0 +1,240 @@
+	"github.com/stellar/go/txnbuild"
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/wallet-backend/internal/signing"
+	"github.com/stellar/wallet-backend/internal/tss"
+	tssErr "github.com/stellar/wallet-backend/internal/tss/errors"
+)
+
+var (
+	RpcPost                 = http.Post
+	UnMarshalRPCResponse    = io.ReadAll
+	UnMarshalJSON           = parseJSONBody
+	callRPC                 = sendRPCRequest
+	UnMarshalErrorResultXdr = parseErrorResultXdr
+)
+
+type transactionService struct {
+	DistributionAccountSignatureClient signing.SignatureClient
+	ChannelAccountSignatureClient      signing.SignatureClient
+	HorizonClient                      horizonclient.ClientInterface
+	RpcUrl                             string
+	BaseFee                            int64
+	Ctx                                context.Context
+}
+
+type TransactionServiceOptions struct {
+	DistributionAccountSignatureClient signing.SignatureClient
+	ChannelAccountSignatureClient      signing.SignatureClient
+	HorizonClient                      horizonclient.ClientInterface
+	RpcUrl                             string
+	BaseFee                            int64
+	Ctx                                context.Context
+}
+
+func (o *TransactionServiceOptions) ValidateOptions() error {
+	if o.DistributionAccountSignatureClient == nil {
+		return fmt.Errorf("distribution account signature client cannot be nil")
+	}
+
+	if o.ChannelAccountSignatureClient == nil {
+		return fmt.Errorf("channel account signature client cannot be nil")
+	}
+
+	if o.HorizonClient == nil {
+		return fmt.Errorf("horizon client cannot be nil")
+	}
+
+	if o.RpcUrl == "" {
+		return fmt.Errorf("rpc url cannot be empty")
+	}
+
+	if o.BaseFee < int64(txnbuild.MinBaseFee) {
+		return fmt.Errorf("base fee is lower than the minimum network fee")
+	}
+	return nil
+}
+
+func NewTransactionService(opts TransactionServiceOptions) (TransactionService, error) {
+	if err := opts.ValidateOptions(); err != nil {
+		return nil, err
+	}
+	return &transactionService{
+		DistributionAccountSignatureClient: opts.DistributionAccountSignatureClient,
+		ChannelAccountSignatureClient:      opts.ChannelAccountSignatureClient,
+		HorizonClient:                      opts.HorizonClient,
+		RpcUrl:                             opts.RpcUrl,
+		BaseFee:                            opts.BaseFee,
+		Ctx:                                opts.Ctx,
+	}, nil
+}
+
+func parseJSONBody(body []byte) (map[string]interface{}, error) {
+	var res map[string]interface{}
+	err := json.Unmarshal(body, &res)
+	if err != nil {
+		return nil, fmt.Errorf(err.Error())
+	}
+	return res, nil
+}
+
+func parseErrorResultXdr(errorResultXdr string) (tss.RPCTXCode, error) {
+	errorResult := xdr.TransactionResult{}
+	err := errorResult.UnmarshalBinary([]byte(errorResultXdr))
+
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf("SendTransaction: unable to unmarshal errorResultXdr: %s", errorResultXdr)
```

Comment:
We shouldn't assume here that SendTransaction is the caller function. Errors should be cascaded back to the callers and wrapped by them.
```suggestion
		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf("unable to unmarshal errorResultXdr: %s", errorResultXdr)
```

Reply:
Agreed